src/main/resources/org/jenkinsci/plugins/workflow/libs/LibraryCachingConfiguration/help-excludedVersionsStr.html

```diff
@@ -0,0 +1,3 @@
+<div>
+    Space separated list of versions excluded from caching. Ex: "master release10 release9"
```

Comment:
But there is no possibility here to specify a regular expression.

FWIW my proposal in JENKINS-38992 was to make this automatic—zero configuration. The SCM should know whether or not a given version was deterministic. The current `SCMSource.retrieve(String, TaskListener)` does not suffice since at least in the case of `AbstractGitSCMSource` the resulting `SCMRevision.isDeterministic` will always be true—the method resolves the actual current head of a branch. Thus we would need to extend the `SCMHead` API in `scm-api` to let the SCM source indicate whether the _head_ is deterministic (like a hash, or arguably a tag) vs. nondeterministic (like a branch). This would be easy to implement in `git` at least.

Reply:
I agree that it should support regex and maybe add an include field as well (only include a single version) but is it really user friendly to cache some versions but not others. In our use case, it would be annoying to cache tags by default because we use semantic versioning and publish minor versions as well as patch versions (ex: release v4.2.1 also releases v4.2 and v4).

All I'm saying is I'm not a fan of hidden behavior and the current implementation is very simple. Is it really necessary to complicate it?